### ISSUE TITLE
Upgrade chef-client to at least 12.5.1

### ISSUE THREAD

**Issue by kamidzi** [bloomberg/chef-bcpc#953]
It's about time. Plus, this update http://www.rubydoc.info/gems/chef/12.5.1/Chef/Mixin/Template/TemplateContext is useful for #939

**Comment by kamidzi** [comment]
Upgrading to anything 12.4.0+ (tested 12.4.0, 12.5.1, 12.6.0) seems to trigger the following:

```
* directory[/var/run/ceph/guests/] action create
    * cannot determine user id for 'libvirt-qemu', does the user exist on this system?
    ================================================================================
    Error executing action `create` on resource 'directory[/var/run/ceph/guests/]'
    ================================================================================

    Chef::Exceptions::UserIDNotFound
    --------------------------------
    cannot determine user id for 'libvirt-qemu', does the user exist on this system?

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/bcpc/recipes/ceph-common.rb

     92: directory "/var/run/ceph/guests/" do
     93:   owner "libvirt-qemu"
     94:   group "libvirtd"
     95:   mode  "0755"
     96: end
     97:
```

**Comment by kamidzi** [comment]
FWIW, seem likes the combo of chef-server 12.3.0 / chef-client 12.5.1 will take

**Comment by kamidzi** [comment]
Additionally, chef-server 12.3.0 plays nicely with chef-client 12.3.0+, though it would probably make little sense to run a client < 12.5.1, as that is the version of the internal client shipping with chef-server 12.3.0

**Comment by erhudy** [comment]
Resolved by #1008.
